catalog/nsili/catalog-nsili-common/src/main/java/org/codice/alliance/nsili/common/ResultDAGConverter.java

```diff
@@ -272,136 +281,160 @@ public static DAG convertResult(Result result, ORB orb, POA poa, List<String> re
         }
 
+        if (shouldAdd(buildAttr(attribute, NsiliConstants.PUBLISHER), resultAttributes)) {
+            Attribute publisherAttr = metacard.getAttribute(Contact.PUBLISHER_NAME);
+            if (publisherAttr != null) {
+                String publisherStr = String.valueOf(publisherAttr.getValue());
+                if (publisherStr != null) {
+                    addStringAttribute(graph,
+                            cardNode,
+                            NsiliConstants.PUBLISHER,
+                            publisherStr,
+                            orb);
+                    addedAttributes.add(buildAttr(attribute, NsiliConstants.PUBLISHER));
+                }
+            }
+        }
+
         return addedAttributes;
     }
 
     public static List<String> addFileNodeWithAttributes(DirectedAcyclicGraph<Node, Edge> graph,
             Node productNode, Metacard metacard, ORB orb, String parentAttrName,
             List<String> resultAttributes) {
+
         List<String> addedAttributes = new ArrayList<>();
-        Any any = orb.create_any();
-        Node fileNode = new Node(0, NodeType.ENTITY_NODE, NsiliConstants.NSIL_FILE, any);
-        graph.addVertex(fileNode);
-        graph.addEdge(productNode, fileNode);
 
-        String attribute = parentAttrName + NsiliConstants.NSIL_FILE;
+        if (metacard.getAttribute(Core.RESOURCE_DOWNLOAD_URL) != null) {
+            Any any = orb.create_any();
+            Node fileNode = new Node(0, NodeType.ENTITY_NODE, NsiliConstants.NSIL_FILE, any);
+            graph.addVertex(fileNode);
+            graph.addEdge(productNode, fileNode);
 
-        if (shouldAdd(buildAttr(attribute, NsiliConstants.ARCHIVED), resultAttributes)) {
-            addBooleanAttribute(graph, fileNode, NsiliConstants.ARCHIVED, false, orb);
-            addedAttributes.add(buildAttr(attribute, NsiliConstants.ARCHIVED));
-        }
+            String attribute = parentAttrName + NsiliConstants.NSIL_FILE;
 
-        if (shouldAdd(buildAttr(attribute, NsiliConstants.CREATOR), resultAttributes)) {
-            Attribute pocAttr = metacard.getAttribute(Metacard.POINT_OF_CONTACT);
-            if (pocAttr != null) {
-                String pocString = String.valueOf(pocAttr.getValue());
-                if (StringUtils.isNotBlank(pocString)) {
-                    addStringAttribute(graph, fileNode, NsiliConstants.CREATOR, pocString, orb);
+            if (shouldAdd(buildAttr(attribute, NsiliConstants.TITLE), resultAttributes)
+                    && metacard.getTitle() != null) {
+                addStringAttribute(graph, fileNode, NsiliConstants.TITLE, metacard.getTitle(), orb);
+                addedAttributes.add(buildAttr(attribute, NsiliConstants.TITLE));
+            }
+
+            if (shouldAdd(buildAttr(attribute, NsiliConstants.PRODUCT_URL), resultAttributes)) {
+                Attribute downloadUrlAttr = metacard.getAttribute(Core.RESOURCE_DOWNLOAD_URL);
```

Comment:
Cleaned up this handling. Also removed a 2nd != null check that was unnecessary.